src/phase0/helpers.rs

```diff
@@ -97,48 +97,40 @@ pub fn is_valid_indexed_attestation<
         ));
     }
 
-    let is_sorted = attesting_indices
-        .windows(2)
-        .map(|pair| {
-            let a = &pair[0];
-            let b = &pair[1];
-            a < b
-        })
-        .all(|x| x);
-    if !is_sorted {
-        return Err(invalid_operation_error(
-            InvalidOperation::IndexedAttestation(
-                InvalidIndexedAttestation::AttestingIndicesNotSorted,
-            ),
-        ));
-    }
-
-    let indices: HashSet<usize> = HashSet::from_iter(attesting_indices.iter().cloned());
-    if indices.len() != indexed_attestation.attesting_indices.len() {
-        let mut seen = HashSet::new();
-        let mut duplicates = vec![];
-        for i in indices.iter() {
-            if seen.contains(i) {
-                duplicates.push(*i);
-            } else {
-                seen.insert(i);
-            }
+    // List of indices is non-empty given check above. Begin iteration from one because a list with
+    // a single entry is "sorted" and contains no duplicates.
+    let mut prev = attesting_indices[0];
+    let mut duplicates = HashSet::new();
+    for index in attesting_indices.as_slice()[1..].iter() {
+        if *index < prev {
+            return Err(invalid_operation_error(
+                InvalidOperation::IndexedAttestation(
+                    InvalidIndexedAttestation::AttestingIndicesNotSorted,
+                ),
+            ));
+        }
+        if *index == prev {
+            duplicates.insert(*index);
```

Comment:
I chose not to short-circuit here because the `DuplicateIndices` error holds the list of duplicates, which I assume should contain all duplicate indices, not just the first duplicate